lib/site_prism/element_container.rb

```diff
@@ -85,6 +85,16 @@ def raise_wait_for_no_if_failed(obj, name, timeout, failed)
             "Timed out after #{timeout}s waiting for no #{obj.class}##{name}"
     end
 
+    def merge_args(find_args, runtime_args, override_options = {})
+      find_args = find_args.dup
+      runtime_args = runtime_args.dup
```

Comment:
When you duplicate these, won't they self-overwrite as they have the same name?

Reply:
No - at first `find_args` refers to the passed in array.  If one modified that array (via pop) it would modify the array in the scope that called `merge_args`.  By calling `dup` on that array, `find_args` in the `merge_args` scope now refers to a new array and can modify it at will without affecting the array from the outer scope.

Reply:
This is really interesting. But I do feel it's complex. But thanks for the explanation. That does make sense!

So if you pass in [1,2,3,4] and then duplicate it. When you call [1,2,3,4].pop, the value of the array after this method has completed is still [1,2,3,4] just the array inside the method would be [1,2,3] right?